modules/ROOT/pages/6.4.1-release-notes.adoc

```diff
@@ -0,0 +1,1334 @@
+
+To prevent this from happening, {productname} made improvements to the Template plugin that now sanitises content when retrieving preview content, and again before running `editor.dom.create` on it during the template insertion action. By doing this, all preview content will now match the template when it is inserted into the editor, which removed the possibility of any XSS issues arising.
+
+In {productname} 6.4.1, Templates are now parsed before preview and insertion to make preview consistent with inserted template content and prevent XSS from occuring.
+
+=== Pressing backspace in an empty line now preserves formatting in a previous empty line
+//#TINY-9454
+
+In previous versions of {productname}, when pressing backspace in an empty line with another empty line containing inline formats above it, the deletion handler was found to be first removing both lines and then inserting a newline. As a consequence, the existing inline formats of the previous empty line was not preserved.
+
+In {productname} 6.4.1, pressing backspace in an empty line now preserves the inline formats of the previous empty line.
+
+=== Pressing enter inside the `fontsizeinput` input would not move the focus back into the editor content
+//#TINY-9598
+
+During development for the new `fontsizeinput` toolbar item, an issue was found that by pressing enter inside the input field would not move the `focus` back into the {productname} editor body. 
+
+As a consequence, any changes made to the font size within the new toolbar input field, would result in the user having no way to use the new font size within the editor.
+
+After some improvements to the `fontsizeinput` toolbar item, pressing enter in the input field now sets the focus back to the {productname} editor viewport, which allows the new font size to be used. 
+
+In {productname} 6.4.1, pressing enter inside the `fontsizeinput` input now returns the focus back to the editor content body.
+
+
+[[additions]]
+== Additions
+
+{productname} 6.4.1 also includes the following additions:
+
+=== New tree component that can be used in dialog body panel
+//#TINY-9532
+
+A new tree component has been created to support the *Advanced Template* premium plugin that works in conjunction with a new tree dialog body panel.
+
+This tree component is limited to *one* level and is used in the dialog body panel to allow users to organize templates into levels for better UI. The tree component is designed to create a visual representation of the hierarchical structure for the template system, which allows users to easily identify the various templates and levels within each catagory, giving users the ability to quickly and easily navigate through the templates, and select the right one for their needs.
+
+This new tree component is an important addition to the new Premium Plugin: *Advanced Templates*, which is now available as a premium plugin with editors running {productname} 6.4.1 or later.
```

Comment:
```suggestion
=== New tree component that can be used in dialog body panel
//#TINY-9532

A new tree component has been added to {productname}, that can be used in custom plugins and models.

The new tree component is designed to create a visual representation of the hierarchical structure of items such as, showcasing folders and the files within these folders, similar to how a directory works which allows users to easily identify various structure levels within set categories, such as how the new `advtemplate` works.

This new tree component is an important addition to {productname} which will be used in several features with up and coming releases in the future.
```

Reply:
@hamza0867 I found the suggested changes, they were outdated and not visible.

Reply:
This one sounds good to me